src/Data/Parameterized/Vector.hs

```diff
@@ -105,9 +105,9 @@ import Numeric.Natural
 import qualified Data.Parameterized.Context as Ctx
 import Data.Parameterized.Utils.Endian
 
--- | Fixed-size non-empty vectors.
+-- | Fixed-size vectors.
 data Vector n a where
```

Comment:
A good point, but in this case, I think we could simply abandon the strict-data definition in favor of a newtype definition. This would entail minor differences in evaluation when you match on the `Vector` constructor, but I don't think these differences would manifest much in practice, since pretty much all pattern matches on `Vector` end up strictly using the underlying vector anyway.

Reply:
Makes sense to me.